dev/Repeater/SelectionModelChildrenRequestedEventArgs.h

```diff
@@ -9,17 +9,19 @@ class SelectionModelChildrenRequestedEventArgs :
     public ReferenceTracker<SelectionModelChildrenRequestedEventArgs, winrt::implementation::SelectionModelChildrenRequestedEventArgsT, winrt::composable, winrt::composing>
 {
 public:
-    SelectionModelChildrenRequestedEventArgs(const winrt::IInspectable& data);
+    SelectionModelChildrenRequestedEventArgs(const winrt::IInspectable& data, SelectionNode* sourceNode);
 
 #pragma region ISelectionModelChildrenRequestedEventArgs
     winrt::IInspectable Source();
+    winrt::IndexPath SourceIndex();
     winrt::IInspectable Children();
     void Children(winrt::IInspectable const& value);
 #pragma endregion
 
-    void Initialize(const winrt::IInspectable& source);
+    void Initialize(const winrt::IInspectable& source, SelectionNode* sourceNode);
 
 private:
     tracker_ref<winrt::IInspectable> m_source{ this };
+    SelectionNode* m_sourceNode{ nullptr };
```

Comment:
>SelectionNode [](start = 4, length = 13)

Is a raw pointer here really the right choice? Maybe a com_ptr would be better

Reply:
This is not a com object. The object is owned during this call so it should be ok. I tried to create a weak_ptr/shared_ptr, but I'm not sure how to do that from this*. enabled_shared_from_this requires atleast one shared ptr for the object to work. I'll leave it as it is for now until i figure out a way to aviod this raw pointer.

Reply:
Agree that we can't guarantee the object doesn't outlive the pointer. What is the type of SelectionNode? Looking around the codebase i see a lot of `shared_ptr<SelectionNode>` references, why can't this just be a shared_ptr?

Reply:
Switched to a weak_ptr.